template_app/src/routes/+page.svelte

```diff
@@ -1,23 +1,78 @@
 <script lang="ts">
-	import { MapLibre } from 'svelte-maplibre';
-	import {SimpleComponent}  from 'ua-components';
-</script>
+  import {
+    Layout,
+    mapContents,
+    sidebarContents,
+  } from "ua-components/two_column_layout";
+  import { MapLibre } from "svelte-maplibre";
+  import TitleMode from "./TitleMode.svelte";
+  import TriangleMode from "./TriangleMode.svelte";
+  import { map as mapStore, backend, mode } from "./globals";
+  import workerWrapper from "./worker?worker";
+  import { type Backend } from "./worker";
+  import * as Comlink from "comlink";
+  import { onMount } from "svelte";
+  import type { Map } from "maplibre-gl";
+
+  // Everythign in this script section is boilerplate; you can ignore it
+
+  onMount(async () => {
+    // If you get "import declarations may only appear at top level of a
+    // module", then you need a newer browser.
+    // https://caniuse.com/mdn-api_worker_worker_ecmascript_modules
+    //
+    // In Firefox 112, go to about:config and enable dom.workers.modules.enabled
+    //
+    // Note this should work fine in older browsers when doing 'npm run build'.
+    // It's only a problem during local dev mode.
+    interface WorkerConstructor {
+      new (): Backend;
+    }
 
-<h1>Welcome to SvelteKit</h1>
-<p>Visit <a href="https://kit.svelte.dev">kit.svelte.dev</a> to read the documentation</p>
+    const MyWorker: Comlink.Remote<WorkerConstructor> = Comlink.wrap(
+      new workerWrapper(),
+    );
+    let backendWorker = await new MyWorker();
+    backend.set(backendWorker);
+  });
```

Comment:
We might want to wrap this in a function that can be imported as a utility which constructs the worker once  as a singleton and allows that singleton to be imported and used at multiple places in the application. Or perhaps a store that gives access to it?

Reply:
It's parameterized by the imported worker (which uses some vite magic), the backend type, and a store for actually using the backend... I'll leave this as TODO. I agree it's confusing-looking boilerplate that users shouldn't see, but I'm not entirely sure how to make it generic yet